idaes/models_extra/gas_solid_contactors/unit_models/tests/test_mb_performance.py

```diff
@@ -0,0 +1,242 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+"""
+
+import pytest
+
+import pyomo.environ as pyo
+from pyomo.dae.flatten import flatten_dae_components
+from pyomo.environ import (
+    ConcreteModel,
+    check_optimal_termination,
+    SolverStatus,
+    value,
+    Var,
+    Constraint,
+)
+from pyomo.util.check_units import assert_units_consistent
+from pyomo.common.config import ConfigBlock
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+from idaes.core import (
+    FlowsheetBlock,
+    MaterialBalanceType,
+    EnergyBalanceType,
+    MomentumBalanceType,
+)
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+    unused_variables_set,
+    large_residuals_set,
+)
+from idaes.core.util.testing import initialization_tester
+from idaes.core.util import scaling as iscale
+from idaes.core.solvers import get_solver
+from idaes.core.util.exceptions import InitializationError
+
+# Import MBR unit model
+from idaes.models_extra.gas_solid_contactors.unit_models.moving_bed import MBR
+
+# Import property packages
+from idaes.models_extra.gas_solid_contactors.properties.methane_iron_OC_reduction.gas_phase_thermo import (
+    GasPhaseParameterBlock,
+)
+from idaes.models_extra.gas_solid_contactors.properties.methane_iron_OC_reduction.solid_phase_thermo import (
+    SolidPhaseParameterBlock,
+)
+from idaes.models_extra.gas_solid_contactors.properties.methane_iron_OC_reduction.hetero_reactions import (
+    HeteroReactionParameterBlock,
+)
+
+
+def build_model(
```

Comment:
In the vein of avoid duplicated code and have a set of standard test cases we use for multiple jobs, could this code be incorporated into the existing MB test cases (e.g. as fixtures for the existing tests).

That way we ensure we are using the same test case for model structure and solution and for performance quantification.

Reply:
I believe I could refactor the existing tests to use this code, although the existing tests would currently only test a subset of the functionality (steady state construction and initialization).

Reply:
I think that is fine, my main aim was to make sure we only had to update one set of code if the MB model changes in future.